### ISSUE TITLE
A11y_mssql for VSCode_mssql Extension_Connections_ScreenReader: Screen Reader is not reading the full information related to the database name.

### ISSUE THREAD

**Issue by Yashi1201** [microsoft/vscode-mssql#17204]
## Actual:
Screen Reader is not reading the full information related to the database name.

**Similar issue is also observed in following:-**

1. mssql Extension-Context Menu actions:- Screen Reader is not reading full information about the System View selected. Screen reader is not reading any information about the more items present for selected system view. 

## Expected:
Screen Reader should read the full information related to the database name. It should also read like press "Shift+F10" to expand the more items.

**GitHubTags**:#A11yMAS;#A11yTCS;#BM_mssql for VSCode_Jan2022;#mssql for VSCode;#Win32;#FTP;#DesktopApp;#A11ySev2;#AINotInScope;#WCAG1.3.1;#NVDA;#Jaws;#STP;#Element:menuitem;

## Environment Details:
Application Name: mssql for VSCode
Windows Version: Win10
Screen Reader: NVDA(2021.3), Jaws(2022.2112.24)

## Repro Steps:

1. Open Visual studio code
2. Tab till extension button and search for "mssql"
3. Tab till install and move to the SQL Server button
4. Tab till Add Connection Button and insert the valid credentials
5. Once the database is added.
6. Run NVDA and observe if it reads anything about the more items present.

## User Impact:
Screen reader users will not get full information about the more items present for database name.

“[Check out Accessibility Insights!](url) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**Comment by smartguest** [comment]
A similar issue has been observed with vscode in general where the screen reader won't get further details about more items present in a collapsed dropdown, this is not just limited to the database tab. Again, this needs to be investigated by the vscode team.

**Comment by smartguest** [comment]
Closing as this is considered default behavior of vscode folders and the screen reader already announces that there are more items inside (expanded/collapsed), which means a user can go inside and read about more contents inside the folder/database if they so wish.

**Comment by Yashi1201** [comment]
@smartguest This bug can be closed by trusted testers by C + AI team after bug verification. Can you please change the bug to open state.
As per your comment, "The screen reader already announces that there are more items inside (expanded/collapsed), which means a user can go inside and read about more contents inside the folder/database if they so wish." The items we see after expansion are not the same when we right click on the parent item. Hence, the key for right click needs to be announced if you state this bug is external to another product, please let us know so that we can take appropriate action.

**Comment by Yashi1201** [comment]
@smartguest Can you please look into the above comment and change the bug state to open state. Thanks

**Comment by Yashi1201** [comment]
@smartguest Could you pls reopen the bug as the issue is still repro. The screenreader is not announcing any information about the more items menu which opens up when we right click with mouse or on pressing keyboard key "Shift+F10". Hence, the key for right click needs to be announced if you state this bug is external to another product, please let us know so that we can take appropriate action.
![17204_issue repro](https://user-images.githubusercontent.com/93374043/216336319-e8b74d88-edc7-4e2d-8206-6feafcb49880.png)

**Comment by smartguest** [comment]
@Yashi1201 This is part of an external issue with VSCODE folders and not with vscode-mssql specifically, they have explicitly stated that they won't be adding this feature to any folder including our server icons due to it adding too much noise for the user experience: https://github.com/microsoft/vscode/issues/141316#issuecomment-1023076548

**Comment by smartguest** [comment]
@alanrenmsft Should we attempt to do anything with this or not? This is a VSCODE dependency issue first and foremost involving the folders. I'm going to file an issue with VSCODE about this.

**Comment by alanrenmsft** [comment]
@smartguest i will take care of the issues

**Comment by alanrenmsft** [comment]
@Yashi1201 
This is what I have observed:
1. Select the connection node
2. Press Shift+F10 or the context menu key, NVDA announces: 'Menu'
3. Press up/down keys to visit the menu items, NVDA will announce the menu item name, the order as well as the total menu item count.

things are all working as I would expect. please let me know what is not working?

![image](https://user-images.githubusercontent.com/13777222/218589139-953bea28-01c9-4030-9618-e604da5e4424.png)
![image](https://user-images.githubusercontent.com/13777222/218589193-30a870b9-e7b4-4267-ad44-c7e4ea058984.png)

**Comment by alanrenmsft** [comment]
@Roopesh-T I suppose this was reactivated as per your team's request, I couldn't find the alias of the person who opened this issue originally. Could you please also take a look at by reply above?

**Comment by Yashi1201** [comment]
@alanrenmsft  Here, when focus is on connection node and screen reader reads "level 1 sqltools2019-3,<default>(adstest) expanded 1 of 1" but screen reader does not read any information about the key of right click to open the menu. So, the screen reader users will not get to know that "Shift+f10" needs to be pressed to open the menu. 

Hence, the key for right click i.e. "Shift+f10" needs to be announced so that screen reader users will know which key to be pressed to get the menu.

Here, is my alias v-yagrawal@microsoft.com

**Comment by alanrenmsft** [comment]
@Yashi1201 Now I understand what you are referring to. But that doesn't seem reasonable to me, it would also be annoying to announce that every time a node is selected. Do you have an example of any application implementing this requirement? I tried on Windows File Explorer, it also doesn't announce the context menu available. 

I think anyone who relies on keyboard only interaction should know to use Shift+F10 or the context menu key to open context menus.

**Comment by Yashi1201** [comment]
@alanrenmsft We discussed this bug with usability team , they said it is default behaviour and the user know the key for right-click to open context menu.

**GithubTags**:#Closed;
